stat/distmat/permutation_test.go

```diff
@@ -21,38 +20,33 @@ func TestUniformPermutation(t *testing.T) {
 			t.Error("Matrix failed")
 		}
 		up.PermTo(m)
+		// Ensure Dims() unchanged.
 		r, c := m.Dims()
 		if r != n || c != n {
-			t.Error("got back matrix of wrong size")
+			t.Errorf("got back matrix of wrong size. expected %d x %d, got %d x %d", n, n, r, c)
+		}
```

Comment:
This test is really not necessary; we can see that `PermTo` does not alter the dimensions of `dst`. Similarly the check for a nil `m` is not needed; `NewDense` cannot fail without a panic.

Please delete L19-27; the extra lines don't add safety but they do distract from the substance of the test.